packages/backend/src/helpers/get-basic-transfer-details.ts

```diff
@@ -0,0 +1,37 @@
+import { IGlobalVariable, ITransferDetails } from '@plumber/types'
+
+/**
+ * Note that this is for every app connection that just needs to show the connection name
+ * without any further instructions (phase 1)
+ * No connection is selected will show up when the app is selected but the user either
+ * 1. Has not added a connection
+ * 2. Managed to remove the connection (most likely through 1 successful pipe transfer)
+ */
+
+export function getEmptyConnectionDetails(
+  position: number,
+  appName: string,
+): ITransferDetails {
+  return {
+    position,
+    appName,
+    connectionName: 'No connection is selected',
+  }
+}
+
+async function getTransferDetails(
+  $: IGlobalVariable,
+): Promise<ITransferDetails | null> {
+  const screenName = $.auth.data?.screenName as string
+  if (!screenName) {
```

Comment:
nit: i can't remember if vault workspace is the only legacy app without `screenName`.

If we want to be more careful, we can also include a check that `$.auth.data` is empty before declaring that no connection is selected.

```
Object.keys($.auth.data ?? {}).length === 0
```